src/actions.rs

```diff
@@ -100,6 +101,22 @@ impl ActionHandler {
                                 continue;
                             }
                             let span = message.spans[0].rls_span().zero_indexed();
+
+                            // build a more sophisticated error message
+                            let mut message_text = message.message.clone();
+                            for sp in &message.spans {
+                                if sp.is_primary && sp.label.is_some() {
+                                    message_text.push_str("\n⇒ ");
+                                    message_text.push_str(sp.label.as_ref().unwrap());
+                                }
+                            }
+                            if !message.children.is_empty() {
+                                message_text.push_str("\n");
```

Comment:
I think if `message.message` can be an empty string this is a bigger problem. Because then users would be presented with a error/warning but without any explanation in the current system. So I think it is save to assume that every compiler error has this set to something non-empty.